openedx/core/lib/xblock_utils/__init__.py

```diff
@@ -92,6 +92,9 @@ def wrap_xblock(
     data = {}
     data.update(extra_data)
 
+    if context:
+        data.update(context.get('wrap_xblock_data', {}))
+
```

Comment:
I like the way this lets us put more of the state directly on the completable xblocks without requiring XBlock authors to be explicitly aware of how completion works, and while minimizing the specialness of verticals. :+1: